packages/twenty-front/src/modules/action-menu/actions/record-actions/single-record/hooks/useExportNoteAction.ts

```diff
@@ -0,0 +1,76 @@
+import { SingleRecordActionHookWithObjectMetadataItem } from '@/action-menu/actions/types/SingleRecordActionHook';
+import { recordStoreFamilyState } from '@/object-record/record-store/states/recordStoreFamilyState';
+import { BlockNoteEditor } from '@blocknote/core';
+import {
+  DOCXExporter,
+  docxDefaultSchemaMappings,
+} from '@blocknote/xl-docx-exporter';
+import {
+  PDFExporter,
+  pdfDefaultSchemaMappings,
+} from '@blocknote/xl-pdf-exporter';
+import * as ReactPDF from '@react-pdf/renderer';
+import { Buffer } from 'buffer';
+import { Packer } from 'docx';
+import { saveAs } from 'file-saver';
+import { useRecoilValue } from 'recoil';
+import { isDefined } from 'twenty-ui';
+
+// Polyfill needed for exportNoteToDocX
+if (typeof window !== 'undefined') {
```

Comment:
I'm a bit worried about all the new dependencies we're adding (my node_module did +66mo when checking out the PR). We should look into the impact on performance and see if we can differ loading (see `lazy(() =>` in code)

Reply:
Most of the dependencies are necessary to enable the export functionality. Many of these were already part of the project, and I’ve primarily updated them as needed.

The new packages introduced are:
- `@blocknote/xl-docx-exporter`
- `@blocknote/xl-pdf-exporter`
- `@react-pdf/renderer`
- `docx`

The only package I added outside of those mentioned in the Blocknote documentation is `file-saver`, which handles downloading the PDF in a more streamlined manner.

That said, I understand the importance of minimizing the performance impact. I agree that lazy loading makes sense here, and I’ll investigate and implement it.

Reply:
In our case, I think using dynamic import() is more appropriate than React.lazy because we are dynamically loading utility functions within event handlers rather than React components.

Reply:
Resolved in recent commit.

Reply:
Perfect!